app/actions/remote/scheduled_post.test.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import DatabaseManager from '@database/manager';
+import NetworkManager from '@managers/network_manager';
+import {getConfigValue} from '@queries/servers/system';
+
+import {deleteScheduledPost, fetchScheduledPosts} from './scheduled_post';
+
+import type ServerDataOperator from '@database/operator/server_data_operator';
+
+const serverUrl = 'baseHandler.test.com';
+
+const scheduledPostsResponse: FetchScheduledPostsResponse = {
+    directChannels: [],
+    bar: [
+        {
+            id: 'scheduled_post_id',
+            channel_id: 'channel_id',
+            root_id: '',
+            message: 'test scheduled post',
+            scheduled_at: 123,
+            user_id: 'user_id',
+            processed_at: 0,
+            update_at: 456,
+            error_code: '',
+        },
+        {
+            id: 'scheduled_post_id_2',
+            channel_id: 'channel_id',
+            root_id: '',
+            message: 'test scheduled post 2',
+            scheduled_at: 123,
+            user_id: 'user_id',
+            processed_at: 0,
+            update_at: 456,
+            error_code: '',
+        },
+    ],
+};
+
+const throwFunc = () => {
+    throw Error('error');
+};
+
+const mockClient = {
+    getScheduledPostsForTeam: jest.fn(() => Promise.resolve({...scheduledPostsResponse})),
+    deleteScheduledPost: jest.fn((scheduledPostId) => {
+        return Promise.resolve(scheduledPostsResponse.bar.find((post) => post.id === scheduledPostId));
```

Comment:
Curious: when you send deleteScheduledPost, you get the post that you deleted back?

Reply:
Yup, I know this is of no use, but the handler is written in this way (as we batch all the records for create/update/delete actions together and return Promise of Scheduled posts). It causes no harm but comes with benefits to test which record was actually deleted.

Reply:
The delete API returned the scheduledPost object?  But to batch delete, you already have the post, no?

It's not a big deal since the data is probably small (and probably don't do a big chunk of it), just odd that it was designed this way.